src/flash-source-buffer.js

```diff
@@ -182,11 +182,18 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
       removeCuesFromTrack(0, Infinity, this.inbandTextTrack_);
     });
 
+    this.mediaSource_.player_.tech_.on('hls-reset', this.resetHls);
```

Comment:
bind would work too. You'll just have to make sure to put it in a local var and use that for both the `on` and `off` otherwise the references passed to `on` and `off` will be to different functions
```
let onHlsReset = this.resetHls.bind(this);

...tech_.on('hls-reset', onHlsReset);
...tech_.off('hls-reset', onHlsReset);
```